### ISSUE TITLE
GPU process exits unexpectedly when playing protected video in Chrome/Edge

### ISSUE THREAD

**Issue by pcslide** [IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT#294]
### Checklist [README]

- [X] Device is not undervolted nor overclocked
- [X] Device is using the [latest drivers](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/wiki/Intel-driver-updates-and-changelogs)
- [X] Application is not cracked, modded and use the latest patch

### Application [Required]

Chrome/Edge

### Processor / Processor Number [Required]

Ryzen 5500

### Graphic Card [Required]

ASUS DG1

### GPU Driver Version [Required]
- [X] 31.0.101.6332
- [X] 31.0.101.6557
- [X] 31.0.101.6078
- [X] 31.0.101.4887
- [X] 31.0.101.4314
- [ ] 31.0.101.4311
- [ ] 31.0.101.4257
- [X] 31.0.101.4255
- [ ] 31.0.101.4148
- [ ] 31.0.101.4146
- [ ] 31.0.101.4125
- [ ] 31.0.101.4123
- [ ] 31.0.101.4091
- [ ] 31.0.101.4090
- [ ] 31.0.101.4034
- [ ] 31.0.101.4032
- [ ] 31.0.101.3975
- [ ] 31.0.101.3959
- [ ] 31.0.101.3802

### Rendering API [Required]

- [ ] Vulkan
- [ ] OpenGL
- [ ] DirectX12
- [ ] DirectX11
- [ ] DirectX10
- [ ] DirectX9
- [X] Not applicable

### Windows Build Number [Required]

- [ ] Windows 11 22H2
- [ ] Windows 11 21H2
- [X] Windows 10 22H2
- [ ] Windows 10 21H2
- [ ] Other (Please specify)

### Intel System Support Utility report

[igcit_ssu.txt](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/11333157/igcit_ssu.txt)

### Description and steps to reproduce [Required]

Details/LOGS/Videos and steps to reproduce had already been reported on [Chromium Bugs](https://crbug.com/1435759), and it seems chromium test engineer had a hard time to get their hand on a DG1 card in order to "triage" the issue.
I knew DG1 was not officially supported on AMD platform, however, I wish someone from Intel could at least shed some light on the cause and the nature of the issue.

**Comment by Karen-Intel** [comment]
Hey @pcslide thank you for posting your findings. 

Just to clarify, the graphics card we'll be reviewing is Intel(R) Iris(R) Xe Graphics according to your SSU.

We have a system we can test on and will be posting our results after doing some testing

Talk to you soon

Karen

**Comment by pcslide** [comment]
@Karen-Intel 
It was sold as individual cards in mainland China by GUNNIR and SanFlag/Shengqi（a brand used by ASUS）

**Comment by Karen-Intel** [comment]
Hii @pcslide 
Just FYI, we have not been able to reproduce in DG2 with AMD/Intel processor or in Iris Xe 11th Gen integrated card.
Please upgrade to latest driver [v.4314](https://www.intel.com/content/www/us/en/download/726609/intel-arc-iris-xe-graphics-whql-windows.html) and [DDU version](https://www.wagnardsoft.com/forums/viewtopic.php?t=4491) and let us know if you see any improvement. 

Thanks!

Karen

**Comment by pcslide** [comment]
@Karen-Intel 
The short answer is, with the latest driver, the issue persists.
I did a DDU routine, and then installed latest driver [v.4314](https://www.intel.com/content/www/us/en/download/726609/intel-arc-iris-xe-graphics-whql-windows.html).
It came to my attention, on my laptop, there is a service called 'Intel Content Protection HDCP Service' running, in comparison, there is no such service on my desktop with DG1 installed.  Could that be the cause of the issue?

**Comment by Karen-Intel** [comment]
Hey @pcslide it sounds like that process is not enabled due to your current PC configuration. Here's more information about [that service.  ](https://www.intel.com/content/www/us/en/docs/programmable/683050/21-4-20-0-0/high-bandwidth-digital-content-protection.html#:~:text=HDCP%20is%20a%20copy%20protection,and%20HDCP%20Specification%20version%202.3%20.)
Also, in the video posted in [Chromium Bugs](https://crbug.com/1435759), I can see there is no content protection enabled. When playing a protected video and recording with OBS (like in the video) only the subtitles would have to be visible. The rest of the media content should be black, that's why I think the content being played on the pc is not protected.
Would be good if you could try to test the same card but on an Intel processor and see the difference.

Hope it helps!

Karen

**Comment by pcslide** [comment]
You are absolutely correct about that OBS is not supposed to record protected videos.  It seems content protection is broken on my desktop. I skimmed through iigd_dch_d.inf, AFAIU, Intel(R) Content Protection HDCP Service should be installed alongside the driver, which seems not to be the case for me.

**Comment by pcslide** [comment]
@Karen-Intel 
Given what you told me, I did a clean install of Win11 on another SSD.  The issue was gone right off the bat.  With the gpu driver installed by windows update or with driver [v.4314](https://www.intel.com/content/www/us/en/download/726609/intel-arc-iris-xe-graphics-whql-windows.html),  protected videos play without a glitch.
However, what is more **puzzling** is, I could not find 'Intel Content Protection HDCP Service' in the system.  And driver [v.4314](https://www.intel.com/content/www/us/en/download/726609/intel-arc-iris-xe-graphics-whql-windows.html)  will not install that service.  Yet, content protection seems working, at least, protected video can not be captured by OBS any more. 
![Screenshot 2023-04-28 223246](https://user-images.githubusercontent.com/59794570/235286390-a7893add-add5-4225-870a-84a7a5d116e5.png)

**Comment by pcslide** [comment]
@Karen-Intel 
I did a clean install of win10 to rule out any issue related to my installation. 
 Here are my findings:
| Clean Install Windows Version | Issue playing protected content |
| ----------- | ----------- |
| Win11_22H2 | NO|
| Win10_22H2| **YES** |

**Comment by Karen-Intel** [comment]
@pcslide thank you for all the experiments you've done!
Allow me to do some research internally about this unexpected behavior on Windows 10 and next week I will get back with my findings too 

Have a great weekend

Karen

**Comment by Karen-Intel** [comment]
Hey @pcslide just a quick update.

I'm currently investigating with my contacts overseas to find out if we have access to this card you have. 
Looks like this will take a while so I appreciate your patience in advance. 
Once I have an update I will get back with an update

Talk to you soon

Karen

**Comment by Karen-Intel** [comment]
Hii @pcslide sorry for the delay.

I could finally get my hands on an Iris Xe Max DG1. 
I performed a small testing on Windows 11 using the latest driver v4369 and [I was able to record for you ](https://drive.google.com/drive/folders/1OcTdV25SVLmyY03o8g9p_KzLcds6qtm1?usp=share_link)in the following vid how the content remains protected.

Can you verify on your end if this driver fixes the issue on Win 11 for you? 
Based on your last comment, you couldn't see it protected but in a system with similar configuration, but as you can see in the video, content remains protected on my end. I am using a 12900K CPU + Iris Xe Max Graphics + v4369

Please let me know and thank you for your patience!

Karen

**Comment by pcslide** [comment]
@Karen-Intel 
On Windows 11, the video is protected all along.  It is on Win 10,  I have the issue of GPU process exiting repeatedly which eventually brings down whole GPU hardware acceleration in Chrome/EDGE.   The issue of video being not protected on Win 10 is just a 'sideshow' from my perspective.

**Comment by Karen-Intel** [comment]
@pcslide Gotcha, my mistake. Will update with my findings soon 

Karen

**Comment by Karen-Intel** [comment]
@pcslide hey, sorry for the delay

Just confirmed the issue with latest driver v.4369 using a similar system to yours. 
Issue is also visible in Intel ARC in Windows10
I'm making sure to report this issue asap.
FYI our driver cadence for fixes can take from 3 up to 6 months, however the impact is being shared in our reports and in case we have news I will post them in this thread. Let's keep it open please

Ty for your help!

Karen

**Comment by pcslide** [comment]
@Karen-Intel 
I'm glad you finally confirmed the issue.
If I may ask,
Does this issue only present itself on an AMD platform?
What's Intel's stance regarding DG1 support right now, especially on platform other than Intel's 11/12th gen ?
Do I need to reopen the issue on Chrome bug list? Or will your team take it from here?

**Comment by Karen-Intel** [comment]
Hey @pcslide 
Nope as I was able to reproduce it in a 12th Gen + the DG1 Card
Now for the driver updates, newer products have faster release cadence for fixes, but I will make sure to share the impact with the dev team so they consider to include it in the regular cadence for <11th Gen products.
And about reopening the issue I don't think it's necessary as I was able to confirm it on Edge too so, we will follow it up internally and if necessary we will involve any other 3rd party in our report.

Have a great day!

Karen

**Comment by Karen-Intel** [comment]
Hey @pcslide 
Actually I don't have access to it anymore :( But as far as I know the sleeping states issues are already being looked into.
We have seen quite a big improvement in the latest beta [v.4382.](https://www.intel.com/content/www/us/en/download/729157/intel-arc-iris-xe-graphics-beta-windows.html) Have you tried it yet?

Karen

**Comment by pcslide** [comment]
@Karen-Intel 
I didn't expect you returning the card right away.😯  As you don't have the card anymore, I suppose I will stay away from v.43XX until the sleeping issue is resolved.

**Comment by Karen-Intel** [comment]
@pcslide have you tested this behavior with our latest driver?
The sleeping issue has been resolved a few driver versions back. 

Karen

**Comment by pcslide** [comment]
Firstly, the sample video in issue #351 is not protected at all.
Secondly, IMHO, it is undesirable to disable all GPU related accelerations in chrome just because some video decoding processes failed.  
Thirdly, Intel DG doesn't behave that way in Win11, and for comparison, Nvidia GPU doesn't have the issue in Win10 and Win11.  FYI, Nvidia GPUs don't support hardware decoding for protected videos as well.

Overall, I suppose issue #351 was closed prematurely.

**Comment by pcslide** [comment]
@Karen-Intel   Have you or your colleagues tested on your side?   I would prefer someone from Intel to test it first if possible.  As for why I am so reluctant,  that was because ver.4644(most likely the firmware) eventually broke the hardware relating to DisplayPort of my card(the hardware failed several days after I rolled back to older driver because of issue #440 ).  My reseller send me a brand new card for exchange out of goodwill and I hope I am not breaking it again.

**Comment by Karen-Intel** [comment]
@pcslide yes we have tested it internally. 
About issue #440 related to #433, customers have confirmed v.4885 fixed their DP issues.
About content protection in issue #351 it is all the info I have :(

**Comment by pcslide** [comment]
@Karen-Intel  I tried ver.101.4887, not only did it not solved this GPU process exits unexpectedly issue, it introduced Chrome/Edge startup freezing issue which was reported [on Intel support community ](https://community.intel.com/t5/Intel-ARC-Graphics/%E5%9C%A8%E6%9C%80%E6%96%B0Chrome%E4%B8%8A%E6%AD%BB%E6%9C%BA-%E6%89%80%E6%9C%89%E8%BD%AF%E4%BB%B6%E6%97%A0%E5%93%8D%E5%BA%94/m-p/1530720/emcs_t/S2h8ZW1haWx8dG9waWNfc3Vic2NyaXB0aW9ufExOREIxUTVXM0FSS1BLfDE1MzA3MjB8U1VCU0NSSVBUSU9OU3xoSw#M8614) :(

**Comment by Karen-Intel** [comment]
Hey @pcslide no worries. I will report the GPU process exiting + the freezing in Chrome and Edge. Allow me to get back to the lab and check the Win10 machine, I will share the report number here when its done :)

Karen

**Comment by pcslide** [comment]
@rmorera-intel @TheWonderfulTartiflette @jpovixwm
1. To clarify, that YouTube movie was chosen, because I thought we need a test sample everyone can access.  At that time I was not aware google has this 
https://shaka-player-demo.appspot.com/demo/#audiolang=en-US;textlang=en-US;uilang=en-US;asset=https://storage.googleapis.com/shaka-demo-assets/angel-one-widevine/dash.mpd;panel=ALL_CONTENT;panelData=drm:WIDEVINE;build=uncompiled
2.  What really annoyed me is playing certain video could break GPU hardware acceleration/decoding in Chrome.  And now it seems the issue was solved.  I think we could close this issue after everyone( #351 #566 ) confirms that.

**Comment by jpovixwm** [comment]
@pcslide It's still broken for me in Chromium based browsers (Chromium, Edge, Brave) on Windows 10, exactly as in #351. Also reproducible using your shaka-player-demo [link](https://shaka-player-demo.appspot.com/demo/#audiolang=en-US;textlang=en-US;uilang=en-US;asset=https://storage.googleapis.com/shaka-demo-assets/angel-one-widevine/dash.mpd;panel=ALL_CONTENT;panelData=drm:WIDEVINE;build=uncompiled) in Edge:

https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/assets/53523617/6ddd3bac-a084-44cc-bd56-db08ed90404a

In this video, attempting to playback a video with HW acceleration fails three times, after which **Edge completely deactivates hardware acceleration for the entire browser process**. It has nothing to do with DRM, as it reproduces in exactly the same way with the video in #351 which is 100% DRM-free.

@rmorera-intel
> Our driver debug team has concluded that this is not a defect on the drivers

Then what is it? Is it a chromium bug? Did the driver debug team actually test this with **Windows 10**? As noted by @pcslide in https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/issues/294#issuecomment-1528780089, this issue only manifests in Windows 10 and not in Windows 11.
How is it acceptable to have video playback in a browser cause such second-long blackouts, followed by HW accel being completely disabled by the browser after it notices that HW accel is malfunctioning?
So, to recap: You attempt to play some video three times, HW accel gets automatically disabled by chromium process-wide, and then you're left with a sluggish browser because HW accel is totally a must to get a smooth web browsing experience in this day and age. The only way to restore HW accel in this case is to close down ALL browser windows and relaunch the browser. Extremely annoying if you're using multiple browser windows in your workflow.

@rmorera-intel I am clearly not the only person experiencing this issue. Can you at least clearly tell us whether this is indeed a bug in the Arc driver, a bug in Windows 10 which received a fix in Windows 11, or a bug in the chromium/blink web browser engine? Why does it only occur on Windows 10 while Win 11 is unaffected?

**Comment by pcslide** [comment]
@jpovixwm  
Have you tried latest version of Chrome?  On my side,  GPU process did exit unexpectedly the first time I played video.  However, the second and the third refresh of the video page didn't trigger another GPU process exit.  As a result, HW accel will not break  no matter how many times I refresh the video page.  I would not say that was perfect, but it was acceptable to me.
On Microsoft's Edge, it's a different story,   the issue still  persists.

**Comment by jpovixwm** [comment]
@pcslide Thank you. I tested it with Edge Beta (Microsoft Edge 121.0.2277.65, which uses version 121.0.6167.57 of chromium) and it behaves exactly as you said. First playback causes blackout, but subsequent reloads of the page do not. This is definitely an improvement, but obviously still not perfect.
Out of curiosity, are you still seeing a difference in behavior between Win 11 and Win 10? (I don't really have a way to check myself because I only have Win 10 installed)

**Comment by rmorera-intel** [comment]
@jpovixwm I'll double-check with debug on this, but based on our conclusions and the one in [Chromium ](https://bugs.chromium.org/p/chromium/issues/detail?id=1435759)it would seem to be isolated to an OS issue.

**Comment by SavageWaffle** [comment]
Has this been resolved using the latest driver? Can we resolve this?

**Comment by pcslide** [comment]
@SavageWaffle 
I did a quick test using driver version 101.6078 with latest chrome stable.  I suppose I encounter a degradation either in driver or in chrome.   Now, it seems GPU acceleration will be disabled after two page refresh if a protected video is presented.  We are back at where I started this thread,  and that was almost 2 years ago.

**Comment by Felipe-Intel** [comment]
@rmorera-intel please check

**Comment by SavageWaffle** [comment]
@pcslide Just curious by chance if any change with latest driver they dropped today?

**Comment by SavageWaffle** [comment]
Can we validate if this has been solved in the newest driver that dropped yesterday.

**Comment by pcslide** [comment]
> Can we validate if this has been solved in the newest driver that dropped yesterday.

The short answer is no.
However, it seems the log revealed a little bit more to the issue, although I am not sure it was relevant.  With the newest driver, logs mention "Failed to init HECI driver":

[13616:1380:0131/154609.399:ERROR:registration_request.cc(291)] Registration response error message: DEPRECATED_ENDPOINT
Fri Jan 31 15:46:28 2025: IGSC: (D:\qb\workspace\31779\source\igsc-master\lib\igsc_lib.c:gsc_driver_init():218) Error in HECI init (3)
Fri Jan 31 15:46:28 2025: IGSC: (D:\qb\workspace\31779\source\igsc-master\lib\igsc_lib.c:igsc_device_subsystem_ids():1597) Failed to init HECI driver
[13616:1380:0131/154635.540:ERROR:registration_request.cc(291)] Registration response error message: DEPRECATED_ENDPOINT
[13616:9280:0131/154710.064:ERROR:gpu_process_host.cc(981)] GPU process exited unexpectedly: exit_code=34
Fri Jan 31 15:47:31 2025: IGSC: (D:\qb\workspace\31779\source\igsc-master\lib\igsc_lib.c:gsc_driver_init():218) Error in HECI init (3)
Fri Jan 31 15:47:31 2025: IGSC: (D:\qb\workspace\31779\source\igsc-master\lib\igsc_lib.c:igsc_device_subsystem_ids():1597) Failed to init HECI driver
Fri Jan 31 15:47:34 2025: IGSC: (D:\qb\workspace\31779\source\igsc-master\lib\igsc_lib.c:gsc_driver_init():218) Error in HECI init (3)
Fri Jan 31 15:47:34 2025: IGSC: (D:\qb\workspace\31779\source\igsc-master\lib\igsc_lib.c:igsc_device_subsystem_ids():1597) Failed to init HECI driver
[13616:1380:0131/154734.393:ERROR:registration_request.cc(291)] Registration response error message: DEPRECATED_ENDPOINT
[13616:9280:0131/154737.636:ERROR:gpu_process_host.cc(981)] GPU process exited unexpectedly: exit_code=34
[13616:9280:0131/154752.722:ERROR:gpu_process_host.cc(981)] GPU process exited unexpectedly: exit_code=34
[13616:1380:0131/154914.038:ERROR:registration_request.cc(291)] Registration response error message: DEPRECATED_ENDPOINT

**Comment by rmorera-intel** [comment]
Please download and install new driver [32.0.101.6632](https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html)

**Comment by pcslide** [comment]
> Please download and install new driver [32.0.101.6632](https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html)
"Failed to init HECI driver" is fixed.   However, GPU process still exits unexpectedly when playing protected video.

**Comment by rmorera-intel** [comment]
I apologize for the radio silence on this one. 
I re-tested the original issue with driver 32.0.101.6874 and I am unable to reproduce it. Please check this video and let me know if I am missing anything in my repro steps: https://youtu.be/yf_AwmShQgQ

Consequently please test with our latest driver [32.0.101.6876 ](https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html)and let me know if you still see the issue.

**Comment by rmorera-intel** [comment]
I'm closing this one now based on my previous comment. If you still see the issue please open a new bug report.
